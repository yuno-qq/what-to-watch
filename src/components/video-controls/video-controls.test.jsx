import React from 'react'
import renderer from 'react-test-renderer'
import {describe, test, expect} from "@jest/globals"

import VideoControls from "./video-controls.jsx"


describe(`<VideoControls> snapshots:`, () => {
  test(`should render without direction`, () => {
    const tree = renderer.create(<VideoControls
      isPlaying={true}
      duration={null}
      currentTime={0}
      currentTimeByClick={null}
      dragValue={null}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test(`should render on pause`, () => {
    const tree = renderer.create(<VideoControls
      isPlaying={false}
      duration={null}
      currentTime={0}
      currentTimeByClick={null}
      dragValue={null}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test(`should render with duration`, () => {
    const tree = renderer.create(<VideoControls
      isPlaying={true}
      duration={100}
      currentTime={30}
      currentTimeByClick={null}
      dragValue={null}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test(`should render with dragValue`, () => {
    const tree = renderer.create(<VideoControls
      isPlaying={true}
      duration={100}
      currentTime={30}
      currentTimeByClick={null}
      dragValue={88}
    />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
