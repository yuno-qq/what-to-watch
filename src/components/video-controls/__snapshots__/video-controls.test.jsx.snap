// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoControls> snapshots: should render on pause 1`] = `
<div
  className="player__controls"
>
  <div
    className="player__controls-row"
  >
    <div
      className="player__time"
    >
      <progress
        className="player__progress"
        max="100"
        onClick={[Function]}
        value={0}
      />
      <button
        className="player__toggler"
        onMouseDown={[Function]}
        style={
          {
            "left": "0%",
          }
        }
      >
        Toggler
      </button>
    </div>
    <div
      className="player__time-value"
    >
      --:--
    </div>
  </div>
  <div
    className="player__controls-row"
  >
    <button
      className="player__play"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <div
      className="player__name"
    />
    <button
      className="player__full-screen"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="27"
        viewBox="0 0 27 27"
        width="27"
      >
        <use
          xlinkHref="#full-screen"
        />
      </svg>
      <span>
        Full screen
      </span>
    </button>
  </div>
</div>
`;

exports[`<VideoControls> snapshots: should render with dragValue 1`] = `
<div
  className="player__controls"
>
  <div
    className="player__controls-row"
  >
    <div
      className="player__time"
    >
      <progress
        className="player__progress"
        max="100"
        onClick={[Function]}
        value={88}
      />
      <button
        className="player__toggler"
        onMouseDown={[Function]}
        style={
          {
            "left": "88%",
          }
        }
      >
        Toggler
      </button>
    </div>
    <div
      className="player__time-value"
    >
      01:10
    </div>
  </div>
  <div
    className="player__controls-row"
  >
    <button
      className="player__play"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#pause"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <div
      className="player__name"
    />
    <button
      className="player__full-screen"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="27"
        viewBox="0 0 27 27"
        width="27"
      >
        <use
          xlinkHref="#full-screen"
        />
      </svg>
      <span>
        Full screen
      </span>
    </button>
  </div>
</div>
`;

exports[`<VideoControls> snapshots: should render with duration 1`] = `
<div
  className="player__controls"
>
  <div
    className="player__controls-row"
  >
    <div
      className="player__time"
    >
      <progress
        className="player__progress"
        max="100"
        onClick={[Function]}
        value={30}
      />
      <button
        className="player__toggler"
        onMouseDown={[Function]}
        style={
          {
            "left": "30%",
          }
        }
      >
        Toggler
      </button>
    </div>
    <div
      className="player__time-value"
    >
      01:10
    </div>
  </div>
  <div
    className="player__controls-row"
  >
    <button
      className="player__play"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#pause"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <div
      className="player__name"
    />
    <button
      className="player__full-screen"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="27"
        viewBox="0 0 27 27"
        width="27"
      >
        <use
          xlinkHref="#full-screen"
        />
      </svg>
      <span>
        Full screen
      </span>
    </button>
  </div>
</div>
`;

exports[`<VideoControls> snapshots: should render without direction 1`] = `
<div
  className="player__controls"
>
  <div
    className="player__controls-row"
  >
    <div
      className="player__time"
    >
      <progress
        className="player__progress"
        max="100"
        onClick={[Function]}
        value={0}
      />
      <button
        className="player__toggler"
        onMouseDown={[Function]}
        style={
          {
            "left": "0%",
          }
        }
      >
        Toggler
      </button>
    </div>
    <div
      className="player__time-value"
    >
      --:--
    </div>
  </div>
  <div
    className="player__controls-row"
  >
    <button
      className="player__play"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#pause"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <div
      className="player__name"
    />
    <button
      className="player__full-screen"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="27"
        viewBox="0 0 27 27"
        width="27"
      >
        <use
          xlinkHref="#full-screen"
        />
      </svg>
      <span>
        Full screen
      </span>
    </button>
  </div>
</div>
`;
